tests/averaging/translation/test_translation_averaging_base.py

```diff
@@ -26,6 +30,65 @@ def setUp(self):
 
         self.obj = DummyTranslationAveraging()
 
+    def __execute_test(
+        self, i2Ui1_input: Dict[Tuple[int, int], Unit3], wRi_input: List[Rot3], wti_expected: List[Point3]
+    ) -> None:
+        """Helper function to run the averagaing and assert w/ expected.
+
+        Args:
+            i2Ri1_input (Dict[Tuple[int, int], Rot3]): [description]
+            wRi_expected (List[Rot3]): [description]
+        """
+        if isinstance(self.obj, DummyTranslationAveraging):
+            self.skipTest("Test case invalid for dummy class")
+
+        wti_computed = self.obj.run(len(wRi_input), i2Ui1_input, wRi_input)
+
+        wTi_computed = [Pose3(wRi, wti) for wRi, wti in zip(wRi_input, wti_computed)]
+        wTi_expected = [Pose3(wRi, wti) for wRi, wti in zip(wRi_input, wti_expected)]
+        self.assertTrue(
+            geometry_comparisons.compare_global_poses(
+                wTi_computed, wTi_expected, RELATIVE_ERROR_THRESHOLD, ABSOLUTE_ERROR_THRESHOLD
+            )
+        )
+
+    def test_circle_two_edges(self):
+        """Tests for 4 poses in a circle, with a pose connected to its immediate neighborhood."""
+        wRi_list, i2Ui1_dict, wti_expected = sample_poses.convert_data_for_translation_averaging(
+            sample_poses.CIRCLE_TWO_EDGES_GLOBAL_POSES, sample_poses.CIRCLE_TWO_EDGES_RELATIVE_POSES
+        )
+        self.__execute_test(i2Ui1_dict, wRi_list, wti_expected)
+
+    def test_circle_all_edges(self):
```

Comment:
There are only 4 poses right? I find it easier to think of it as a square (because it is :) ), though lie on a circle. Not sure if you want to rename.

Reply:
If I go to 6 points, this problem will be solved. I mean it will be better as we can always fit a polygon too!